### ISSUE TITLE
Can't seem to get sessions working in aspnet core 2.0

### ISSUE THREAD

**Issue by TehWardy** [aspnet/Home#3187]
I've seen so much information on this it's unreal and it looks dead simple but this weekend I figured I would try converting one of our existing aspnet 4.6 apps over. 

Here's what I have in startup ...

```
using Core.Cryptos;
using Core.Objects;
using Core.Objects.Dtos;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.AspNetCore.Routing;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;

namespace ShellCore
{
    public class Startup
    {
        static IConfiguration config;

        public Startup(IConfiguration configuration)
        {
            config = configuration;
        }
        
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddSingleton(config);
            services.AddSingleton<ICrypto<Signature>>(new AesCrypto<Signature>());
            services.AddSingleton<ICache<PageCacheItem>>(new Cache<PageCacheItem>());
            services.AddDistributedMemoryCache();
            services.AddSession();
            services.AddMvc();
        }
        
        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
        {
            app.UseSession();

            if (env.IsDevelopment())
            {
                app.UseBrowserLink();
                app.UseDeveloperExceptionPage();
            }
            else
                app.UseExceptionHandler("/Home/Error");
            
            app.UseStaticFiles();
            app.UseResponseCaching();
            app.UseMvc(routes =>
            {
                routes.MapRoute("default", "{*path}", new { controller = "Home", action = "Index" });
                routes.MapRoute("page", "{*path}");
            });
        }
    }
}
```

... as soon as I do this in a controller action I get an exception ...

`HttpContext.Session...`

... The exception reads ...

> System.InvalidOperationException: 'Session has not been configured for this application or request.'

... what have i done wrong?

**Comment by Tratcher** [comment]
That's odd, the code looks fine. Is there anything odd in the application logs?

**Comment by TehWardy** [comment]
Ok so it gets more peculiar ... 
Today I opened up the same project on my work pc and without making a single change I built and ran the code without a hitch.

My work machine has had VS 2017 installed on top of win 10 for some time and I always try to take the updates VS comes with when they arrive in the notification.
My home machine that's experiencing this problem is a week old clean install of windows 10 + VS 2017 patched up as needed from the latest installer direct from vs online.

Both are running the enterprise edition so i'm pretty sure it's not a version diff ... 
Which made me think there's some sort of missing dependency in a clean install that's missing or something.

Then i copied the project back to my home machine after deleting the copy I already had there. 
I did a clean on the solution then a full rebuild and run ... the problem is gone.

I'm totally confused and have no idea what fixed it.
Having tried several cleans and rebuilds before moving the project to my work pc I was confident that those steps alone would not fix the problem so I'm baffled as to what it was.
